Userland/Libraries/LibWeb/Crypto/CryptoBindings.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2023, stelar7 <dudedbz@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibWeb/WebIDL/Buffers.h>
+
+// FIXME: Generate these from IDL
+namespace Web::Bindings {
+
+// https://w3c.github.io/webcrypto/#JsonWebKey-dictionary
+struct RsaOtherPrimesInfo {
+    Optional<String> r;
+    Optional<String> d;
+    Optional<String> t;
+};
+
+// https://w3c.github.io/webcrypto/#JsonWebKey-dictionary
+struct JsonWebKey {
+    Optional<String> kty;
+    Optional<String> use;
+    Optional<Vector<String>> key_ops;
+    Optional<String> alg;
+    Optional<bool> ext;
+    Optional<String> crv;
+    Optional<String> x;
+    Optional<String> y;
+    Optional<String> d;
+    Optional<String> n;
+    Optional<String> e;
+    Optional<String> p;
+    Optional<String> q;
+    Optional<String> dp;
+    Optional<String> dq;
+    Optional<String> qi;
+    Optional<Vector<RsaOtherPrimesInfo>> oth;
+    Optional<String> k;
+};
+
+// https://w3c.github.io/webcrypto/#dfn-Algorithm
+struct Algorithm {
+    String name;
+};
+
+// https://w3c.github.io/webcrypto/#key-algorithm-dictionary
+class KeyAlgorithm : public JS::Object {
```

Comment:
Wait, why does this need to be GC allocated? Normally we model dictionaries as raw structs.

Reply:
According to the spec

> The [KeyAlgorithm](https://w3c.github.io/webcrypto/#dfn-KeyAlgorithm) dictionary is provided to aid in documenting how fixed, public properties of a [CryptoKey](https://w3c.github.io/webcrypto/#dfn-CryptoKey) are reflected back to an application. The actual dictionary type is never exposed to applications.

So it sounds like we can make it whatever we like?